operator/pkg/utils/patch/patch.go

```diff
@@ -23,6 +23,23 @@ import (
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
 )
 
+// VolumeClaimTemplateSpec returns the merged VolumeClaimTemplate spes
+func PersistentVolumeClaimSpec(defaultSpec, patch *v1.PersistentVolumeClaimSpec) (v1.PersistentVolumeClaimSpec, error) {
+	if patch == nil {
+		return *defaultSpec, nil
+	}
+	merged, err := mergePatch(defaultSpec, patch, v1.PersistentVolumeClaimSpec{})
```

Comment:
IIUC, sometime we do want to introduce special logic in merging (e.g., PodSpec has this mergeContainerArgs that help merging the arguement instead of replacing one by other). A generic wrapper may not be able to do so without some extra complexity. Maybe when we see more cases when KIT need to expose specs, we can have a better understanding how to create a generic wrapper?